### DISCUSSION TITLE
Magisk module Android not accessible via mitmweb

### DISCUSSION THREAD

**Discussion by eggerand13** [mitmproxy/mitmproxy#6276] [General]
#### Problem Description
When following the excellent instructions around [obtaining the Magisk module form mitmweb](https://github.com/mitmproxy/mitmproxy/blob/main/docs/src/content/howto-install-system-trusted-ca-android.md) I have the problem that the URL http://mitm.it/cert/magisk is not reachable within the Android emulator. 

#### Steps to reproduce the behavior:
Command to start the AVD: `emulator -avd Pixel_4_API_29_with_Google_Play -http-proxy 127.0.0.1:8080`
Magisk is installeda and can grant root access

Output from mitmweb:
```
[22:03:28.055] HTTP(S) proxy listening at *:8080.
[22:03:28.056] Web server listening at http://127.0.0.1:8081/
```

Output in the emulator:
![image](https://github.com/mitmproxy/mitmproxy/assets/25527243/cb3d338b-786f-4757-a109-b119fedc7b6b)

Running on Windows 10

#### System Information
Mitmproxy: 9.0.1 binary
Python:    3.11.0
OpenSSL:   OpenSSL 3.0.7 1 Nov 2022
Platform:  Windows-10-10.0.19045-SP0

Tagging @Jurrie because he is an expert on the matter

Thank you very much your help

**Comment by Jurrie** [comment]
Hi eggerand13,

Judging by the error message 'DNS_PROBE_FINISHED_BAD_CONFIG' I suspect something else is wrong...

I fired up the following:
`./mitmweb`
`emulator -avd Pixel_4_API_30 -http-proxy 127.0.0.1:8080`

When you browse to http://mitm.it, does it show the page saying "Install mitmproxy's Certificate Authority"? It does for me.

Are you able to browse to HTTPS sites? I would expect that this will not work, and will trigger a "Client TLS handshake failed." exception in mitmproxy's log. Is this the case for you?

Are you able to browse to non-SSL sites, like http://neverssl.com? This works for me.

I am using:
```
$ mitmweb --version
Mitmproxy: 10.0.0.dev
Python:    3.10.6
OpenSSL:   OpenSSL 3.0.8 7 Feb 2023
Platform:  Linux-5.19.0-45-generic-x86_64-with-glibc2.35
```

```
$ emulator -version
INFO    | Android emulator version 32.1.11.0 (build_id 9536276) (CL:N/A)
```

For browsing I use the `WebView Browser Tester 83.0.4103.120` (as shown in the top of the screen).

Jurrie

**Comment by eggerand13** [comment]
Hey @Jurrie 

Thank you for the extensive answer and sorry for the rather late reply.

I don't receive anything regarding  "Install mitmproxy's Certificate Authority".

I can actually not access any site (even ones that don't use ssl). I'll receive the same error as in the screenshot above.
When deactivating the proxy setting for the AVD, everything works again as expected.

I get logs like this in mitmweb

```
[21:56:33.906][127.0.0.1:51082] Server TLS handshake failed. Certificate verify failed: IP address mismatch
[21:56:33.907][127.0.0.1:51082] Unable to establish TLS connection with server (Certificate verify failed: IP address mismatch). Trying to establish TLS with client anyway. If you plan to redirect requests away from this server, consider setting `connection_strategy` to `lazy` to suppress early connections.
[21:56:33.913][127.0.0.1:51082] server disconnect 5.9.164.112:853
```

In terms of browser I use the preinstalled Chrome (v 74.0.3729)

**Comment by mhils** [comment]
The error message indicates that mitmproxy is not happy connecting to the upstream server because of certificate verification issues. You can try and see if setting https://docs.mitmproxy.org/stable/concepts-options/#ssl_insecure fixes it.
